config/config_helpers.go

```diff
@@ -255,6 +276,10 @@ func runCommand(args []string, include *IncludeConfig, terragruntOptions *option
 		terragruntOptions.Logger.Debugf("run_cmd output: [%s]", value)
 	}
 
+	runCommandCacheMutex.Lock()
+	runCommandCache[cacheKey] = value
+	runCommandCacheMutex.Unlock()
```

Comment:
Similarly, extract the cache storage logic into a function too, with a similar comment.